src/test/java/io/github/stepio/cache/caffeine/MultiConfigurationCacheManagerCustomCaffeineTest.java

```diff
@@ -60,7 +59,7 @@ public void testCreateNativeCaffeineCacheWithCustomCaches() {
 
         final Object etalon = aCustom;
         await().atMost(300, TimeUnit.MILLISECONDS)
-                .until(() -> this.cachedDataHolder.newCachedCustomObject() != etalon);
```

Comment:
Checking by address highlights that the cache returns exactly the same object. Any particular reason for this change? Same for a few other occurrences below.

Reply:
This is a very hard and coupled assumption. What if the cache would e. g. have a flyweight pattern integrated?
I came across this through IntelliJ's standard code inspections.
Imho equals is fine. However you are more familar with the code - so if you like I would revert the comparisons.